### ISSUE TITLE
Windows: signal.SIGALRM is not compatible on windows.

### ISSUE THREAD

**Issue by spitfiredd** [pywren/pywren-ibm-cloud#121]
It looks like the signal module has limited compatibility with windows, and `signal.SIGALRM` and `signal.alarm` are both only available on *nix based systems.

I am on windows 10, python 3.7.2

I came across,

- https://code.activestate.com/lists/python-list/693666/
- https://stackoverflow.com/questions/8420422/python-windows-equivalent-of-sigalrm

```
(.venv) D:\python\pywren-ibm>python testwren.py
IBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID 553e0bef-efa9
Executor ID 553e0bef-efa9 Uploading function and data - Total: 3.4KiB
Executor ID 553e0bef-efa9 Starting function invocation: hello_world() - Total: 1 activations
Executor ID 553e0bef-efa9 Getting results ...
EIBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID 42d6e074-849e
Executor ID 42d6e074-849e Uploading function and data - Total: 3.8KiB
Executor ID 42d6e074-849e Starting function invocation: simple_map_function() - Total: 4 activations
Executor ID 42d6e074-849e Getting results ...
EIBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID 2b65c1e6-b2e2
Executor ID 2b65c1e6-b2e2 Uploading function and data - Total: 4.2KiB
Executor ID 2b65c1e6-b2e2 Starting function invocation: simple_map_function() - Total: 4 activations
Executor ID 2b65c1e6-b2e2 Uploading function and data - Total: 9.3KiB
Executor ID 2b65c1e6-b2e2 Starting function invocation: simple_reduce_function() - Total: 1 activations
Executor ID 2b65c1e6-b2e2 Getting results ...
EIBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID d63f92b8-ba3e
Executor ID d63f92b8-ba3e Uploading function and data - Total: 4.5KiB
Executor ID d63f92b8-ba3e Starting function invocation: simple_map_function() - Total: 2 activations
Executor ID d63f92b8-ba3e Uploading function and data - Total: 4.5KiB
Executor ID d63f92b8-ba3e Starting function invocation: simple_map_function() - Total: 2 activations
Executor ID d63f92b8-ba3e Getting results ...
EIBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID e34103df-ebcc
Executor ID e34103df-ebcc Uploading function and data - Total: 15.1KiB
Executor ID e34103df-ebcc Starting function invocation: my_map_function_bucket() - Total: 8 activations
Executor ID e34103df-ebcc Uploading function and data - Total: 19.8KiB
Executor ID e34103df-ebcc Starting function invocation: my_reduce_function() - Total: 1 activations
Executor ID e34103df-ebcc Getting results ...
EIBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID cc2c15eb-3cfe
Executor ID cc2c15eb-3cfe Uploading function and data - Total: 15.6KiB
Executor ID cc2c15eb-3cfe Starting function invocation: my_map_function_bucket() - Total: 8 activations
Executor ID cc2c15eb-3cfe Uploading function and data - Total: 32.0KiB
Executor ID cc2c15eb-3cfe Starting function invocation: my_reduce_function() - Total: 6 activations
Executor ID cc2c15eb-3cfe Getting results ...
EIBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID 2fc1066a-1953
Executor ID 2fc1066a-1953 Uploading function and data - Total: 15.7KiB
Executor ID 2fc1066a-1953 Starting function invocation: my_map_function_bucket() - Total: 6 activations
Executor ID 2fc1066a-1953 Uploading function and data - Total: 19.9KiB
Executor ID 2fc1066a-1953 Starting function invocation: my_reduce_function() - Total: 1 activations
Executor ID 2fc1066a-1953 Getting results ...
EIBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID b9a7cae4-63a0
Executor ID b9a7cae4-63a0 Uploading function and data - Total: 16.2KiB
Executor ID b9a7cae4-63a0 Starting function invocation: my_map_function_bucket() - Total: 6 activations
Executor ID b9a7cae4-63a0 Uploading function and data - Total: 32.2KiB
Executor ID b9a7cae4-63a0 Starting function invocation: my_reduce_function() - Total: 6 activations
Executor ID b9a7cae4-63a0 Getting results ...
ERetrieving items' names from bucket: spif-pywren-bucket, prefix: __pywren.test
IBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID 65a2828a-bc0f
Executor ID 65a2828a-bc0f Uploading function and data - Total: 16.6KiB
Executor ID 65a2828a-bc0f Starting function invocation: my_map_function_key() - Total: 6 activations
Executor ID 65a2828a-bc0f Uploading function and data - Total: 20.8KiB
Executor ID 65a2828a-bc0f Starting function invocation: my_reduce_function() - Total: 1 activations
Executor ID 65a2828a-bc0f Getting results ...
ERetrieving items' names from bucket: spif-pywren-bucket, prefix: __pywren.test
IBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID f18fe21b-e595
Executor ID f18fe21b-e595 Uploading function and data - Total: 17.1KiB
Executor ID f18fe21b-e595 Starting function invocation: my_map_function_key() - Total: 6 activations
Executor ID f18fe21b-e595 Uploading function and data - Total: 33.1KiB
Executor ID f18fe21b-e595 Starting function invocation: my_reduce_function() - Total: 6 activations
Executor ID f18fe21b-e595 Getting results ...
EIBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID aa582a99-6927
Executor ID aa582a99-6927 Uploading function and data - Total: 17.6KiB
Executor ID aa582a99-6927 Starting function invocation: my_map_function_url() - Total: 5 activations
Executor ID aa582a99-6927 Uploading function and data - Total: 21.4KiB
Executor ID aa582a99-6927 Starting function invocation: my_reduce_function() - Total: 1 activations
Executor ID aa582a99-6927 Getting results ...
ERetrieving items' names from bucket: spif-pywren-bucket, prefix: __pywren.test
IBM Cloud Functions init for Namespace: spitfiredd@gmail.com_dev
IBM Cloud Functions init for Host: https://us-south.functions.cloud.ibm.com
IBM Cloud Functions init for Runtime: ibmfunctions/action-python-v3.7 - 256MB
IBM Cloud Functions executor created with ID bec79da1-f923
Executor ID bec79da1-f923 Uploading function and data - Total: 16.3KiB
Executor ID bec79da1-f923 Starting function invocation: my_map_function_storage_handler() - Total: 6 activations
Executor ID bec79da1-f923 Uploading function and data - Total: 22.2KiB
Executor ID bec79da1-f923 Starting function invocation: my_reduce_function() - Total: 1 activations
Executor ID bec79da1-f923 Getting results ...
E
======================================================================
ERROR: test_call_async (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 108, in test_call_async
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_map (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 125, in test_map
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_map_reduce (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 132, in test_map_reduce
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_multiple_executions (__main__.TestPywren)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 141, in test_multiple_executions
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_chunks_bucket (__main__.TestPywrenCos)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 317, in test_chunks_bucket
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_chunks_bucket_one_reducer_per_object (__main__.TestPywrenCos)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 325, in test_chunks_bucket_one_reducer_per_object
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_map_reduce_cos_bucket (__main__.TestPywrenCos)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 270, in test_map_reduce_cos_bucket
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_map_reduce_cos_bucket_one_reducer_per_object (__main__.TestPywrenCos)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 277, in test_map_reduce_cos_bucket_one_reducer_per_object
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_map_reduce_cos_key (__main__.TestPywrenCos)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 286, in test_map_reduce_cos_key
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_map_reduce_cos_key_one_reducer_per_object (__main__.TestPywrenCos)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 295, in test_map_reduce_cos_key_one_reducer_per_object
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_map_reduce_url (__main__.TestPywrenCos)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 301, in test_map_reduce_url
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

======================================================================
ERROR: test_storage_handler (__main__.TestPywrenCos)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "testwren.py", line 310, in test_storage_handler
    result = pw.get_result()
  File "D:\python\pywren-ibm\.venv\lib\site-packages\pywren_ibm_cloud\wren.py", line 333, in get_result
    signal.signal(signal.SIGALRM, timeout_handler)
AttributeError: module 'signal' has no attribute 'SIGALRM'

----------------------------------------------------------------------
Ran 12 tests in 18.251s

FAILED (errors=12)
```

**Comment by JosepSampe** [comment]
@spitfiredd @gilv Fixed in #122

**Comment by gilv** [comment]
@JosepSampe thanks. @spitfiredd merged
